nextcord/application_command.py

```diff
@@ -0,0 +1,1266 @@
+                temp["guild_id"] = guild_id
+                ret.append(temp)
+        if self.is_global:
+            ret.append(partial_payload)
+        return ret
+
+    def check_against_raw_payload(self, raw_payload: dict, guild_id: Optional[int]) -> bool:
+        """Checks if self.payload values match with what the given raw payload has.
+
+        This doesn't make sure they are equal. Instead, this checks if all key:value pairs inside of any of our payloads
+        also exist inside of the raw_payload. If there are extra keys inside of the raw_payload that aren't in our
+        payloads, they will be ignored.
+
+        Parameters
+        ----------
+        raw_payload: :class:`dict`
+            Dictionary payload our payloads are compared against.
+        guild_id: Optional[:class:`int`]
+            Guild ID to compare against. If None, it's assumed to be a Global command.
+
+        Returns
+        -------
+        :class:`bool`
+            True if any of our payloads has every key:value pair corresponding with key:value's in the raw_payload,
+            False otherwise.
+        """
+        return self._check_against_raw_payload(raw_payload, guild_id)
+        # TODO: Migrate this over once complete.
+        # if guild_id:
+        #     if self._recursive_item_check(self.get_guild_payload(guild_id), raw_payload):
+        #         return True
+        # else:
+        #     if self._recursive_item_check(self.global_payload, raw_payload):
+        #         return True
+        # return False
+
+    def _check_against_raw_payload(self, raw_payload: dict, guild_id: Optional[int]) -> bool:
+        if guild_id:
+            cmd_payload = self.get_guild_payload(guild_id)
+            if cmd_payload["guild_id"] != int(raw_payload["guild_id"]):
+                return False
+        else:
+            cmd_payload = self.global_payload
+
+        if not check_dictionary_values(cmd_payload, raw_payload, "default_permission", "description", "type", "name"):
+            print("Failed first main check.")
+            return False
+
+        for cmd_option in cmd_payload.get("options", []):
+            found_correct_value = False
+            for raw_option in raw_payload.get("options", []):
+                if cmd_option["name"] == raw_option["name"]:
+                    found_correct_value = True
+                    # At this time, ApplicationCommand options are identical between locally-generated payloads and
+                    # payloads from Discord. If that were to change, switched to a recursive setup and manually
+                    # check_dictionary_values.
+                    if not deep_dictionary_check(cmd_option, raw_option):
+                        return False
+            if not found_correct_value:
+                return False
+        return True
+
+    def reverse_check_against_raw_payload(self, raw_payload: dict, guild_id: Optional[int]) -> bool:
```

Comment:
``guild_id`` is marked optional, but a guild id is required here? Default to ``None``